src/app/shared/mid-price.ts

```diff
@@ -0,0 +1,95 @@
+import { getCurrentPrice } from "@app/prices"
```

Comment:
this is a bad naming for this file, `shared`, `helper`, etc is not good for app layer. if this is related to price then should be in price folder

Reply:
if we really want to do this then should be part of price or dealer service

Reply:
This one's tricky. It's a mix of dealer service and price service where price service is the fallback. Ideally this is just to facilitate our move to the dealer where we're still not sure of dealer reliability so we're including the current status quo (price service) as a fallback.

Once we're confident in the dealer being reliable I think we'll be able to remove this `mid-price` use-case entirely.